src/isolatable-hybrid.store.ts

```diff
@@ -2,6 +2,15 @@ import { HybridConfigurationStore, IAsyncStore, ISyncStore } from '@eppo/js-clie
 
 export type ServingStoreUpdateStrategy = 'always' | 'expired' | 'empty';
 
+/**
+ * Extension of AsyncStore where its aware it's paired with a SyncStore.
+ * Useful when providing custom persistent stores.
+ */
+export interface IPairedAsyncStore<T> extends IAsyncStore<T> {
+  // If available, will be called when HybridConfigurationStore is constructed to provide a rehydration hook for the serving store
+  registerRehydrate?(rehydrate: (entries?: Record<string, T>) => Promise<void>): void;
```

Comment:
Rather than a generic `registerHooks()` type method, I think we can add hooks one top-level method at a time. It won't make the flow control much more complex, and keep things cleaner I think.

Reply:
agreed - we're not looking at needing an expandable hooks/plugin feature